test/CMakeLists.txt

```diff
@@ -3,7 +3,13 @@ include_directories(..)
 add_executable(unittest
     MainTestSuite.cpp BufferTest.cpp MaeBlockTest.cpp MaeParserTest.cpp ReaderTest.cpp WriterTest.cpp UsageDemo.cpp)
 
-target_link_libraries(unittest maeparser Boost::unit_test_framework)
+target_link_libraries(unittest maeparser Boost::unit_test_framework Boost::filesystem)
+
+set(TEST_SAMPLES test.mae test2.maegz )
+file(COPY ${TEST_SAMPLES} DESTINATION ${CMAKE_BINARY_DIR}/test/samples)
```

Comment:
Is it possible to read the files from the source? Do we have access to that path at runtime?

It may even make sense to compile it in (e.g. compile time definition) - the tests don't need to be portable.

This would allow us to delete the whole output directory before running tests, and would make the tests easier to rerun if I'm just changing a test input file (which is a common change for testing a parser!)

Reply:
No, I don't think we have access to those paths at runtime, but we could definitely compile in the path to the references.